src/mcts/search.cc

```diff
@@ -1145,6 +1145,17 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
       }
       return NodeToProcess::Collision(node, depth, collision_limit);
     }
+    // Probably best place to check for two-fold draws consistently.
+    // Depth starts with 1 at root, so real depth is depth - 1.
+    if (node->IsTwoFoldTerminal() && params_.GetTwoFoldDraws()) {
+      if (depth - 1 < node->GetM()) {
```

Comment:
@borg323 do you think it would be better to call `history_.Last().GetPliesSincePrevRepetition()` here?

Reply:
Now that I understand it, I like it the way it is, but I admit it took some explaining to get there.

Reply:
There isn't history, which would need to regenerated to recompute the cycle length

Reply:
OK, thanks for your answers. This seems to be the correct solution then, and we assume that `GetM()` will be available from now on anyway.